src/ray/gcs/gcs_server/gcs_placement_group_manager.cc

```diff
@@ -305,6 +318,11 @@ void GcsPlacementGroupManager::OnPlacementGroupCreationSuccess(
       placement_group->GetPlacementGroupTableData(),
       [this, placement_group_id](Status status) {
         RAY_CHECK_OK(status);
+
+        if (RescheduleIfStillHasUnplacedBundles(placement_group_id)) {
```

Comment:
I think this shouldn't be here, but around here. https://github.com/ray-project/ray/blob/728e2b36d685f1b578c52c5443759071391e39f3/src/ray/gcs/gcs_server/gcs_placement_group_scheduler.cc#L435

Basically when the placement group commit is over, if there's still unplaced bundles (which can happen if the pg state is mutated in the middle of scheduling due to new node dead), we just mark state as rescheduling and call failure_callback